app/main.py

```diff
@@ -2,5 +2,16 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    func_cache = {}
+
+    def wrapper(*args) -> Callable:
+        if args in func_cache:
```

Comment:
Why you check only `args`?
What if function takes `kwargs`?

Reply:
because function takes only **immutable** arguments 
 I think we don't need to check kwargs here